tests/test_FindTest.py

```diff
@@ -709,3 +709,21 @@ def test_custom_tokenizer(self):
         self.run_test_pairs(
             test_pairs, "Custom tokenizer", tokenizers=[tokenizer]
         )
+
+    def test_citation_fullspan(self):
+        example = 'citation number one is Wilson v. Mar. Overseas Corp., 150 F.3d 1, 6-7 ( 1st Cir. 1998); This is different from Commonwealth v. Bauer, 604 A.2d 1098 (Pa.Super. 1992), my second example'
+        extracted = get_citations(example)
+
+        answers = [(0, (23, 86)), (1, (111, 164))]
```

Comment:
I think this is great, but let's follow the conventions in the other tests. 

I think a better name might be test_FullCaseCitation_fullspan() or something because this doesnt work for other cites right?  

Maybe it does.

Reply:
If you could add a docstring with an explanation of the test.

Reply:
Also I think it would be more typical for our tests to have a tuples with a test and an output expected

Reply:
It currently works as follows (according to my understanding, please let me know if I'm wrong):
1. Text is tokenized
2. Citation tokens (core citation text, e.g. '150 F.3d 1') are used for instantiating citations
3. For each citation, we fetch metadata using the `add_metadata` function (find.py, l.149)

the `add_metadata` function depends on subclasses :
- `SupraCitation` does nothing
- `IdCitation` does nothing
- `UnknownCitation` does nothing
- `ShortCaseCitation` does nothing
- `CaseCitation` does nothing
- `FullCaseCitation` calls `add_post_citation` and `add_defendant`
- `FullJournalCitation` calls `add_journal_metadata`
-  `FullLawCitation` calls `add_law_metadata`

The `add_x_metadata`, `add_post_citations` functions use regex to find the metadata. I am using the start() and end() functions to know where the regex found interesting parts.
The `add_defendant` function iterates over words and stops after any StopWordToken. Here, I needed to use the number of characters in each word to get an idea of the span.

I modified the `add_defendant`, `add_journal_metadata`, `add_post_citation` and `add_law_metadata` functions.
According to above, the solution should be working for `FullCaseCitation`, `FullJournalCitation` and `FullLawCitation`.
I will try to include examples of these in the tests as well.

So about the name I would like to keep it as it is, what do you think ? Otherwise, it would be nice to avoid a combination of pascal case and snake case.

I will change the docstring and adopt your tuple approach.